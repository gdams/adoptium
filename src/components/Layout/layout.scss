// Here you can customize bootstrap
$primary: #14003c;
$secondary: #dadada;

@import "~bootstrap/scss/bootstrap";

/* set a max-width for horizontal fluid layout and make it centered */
.main-banner > .container {
  margin-right: auto;
  margin-left: auto;
}

.container:not(.main-banner > .container):not(.header-area > .container):not(.services > .container) {
  max-width: 60rem;
}

body {
  --black4: #cbd4d9;
  --black9: #2c3437;
  --brand4: #c52158;
  --brand1: #14003c;
  --page-pink: #fcf4f6;
  --bg: white;
  --text-normal: #222;
  --color-fill-banner: #9992;
  --color-text-primary: var(--black9);
  --color-text-accent: var(--color-brand-primary);

  .support-matrix th {
    background-color: #aeadad97;
  }
}

.text-pink {
  color: var(--brand4);
}

.halign-center {
  text-align: center;
}

.valign-middle {
  vertical-align: middle;
}

.bg-pink {
  background-color: var(--brand4) !important;
}

.bg-purple {
  background-color: var(--brand1) !important;
}

.bg-grey {
  background-color: rgb(237, 234, 234) !important;
}

.table-secondary {
  --bs-table-bg: #333;
}

body:not(.dark) .dark-mode-only,
.dark .light-mode-only {
  display: none;
}

.hide-on-mobile {
  @media only screen and (width <= 500px) {
    display: none !important;
  }
}

.adopt-demo-container {
  position: relative;
}

.adopt-demo {
  background: #1f272a;
  border-radius: 0.4rem;
  box-shadow: 5px 10px 50px rgba(0, 0, 0, 25%);
  color: #fff;
  height: 450px;
  max-width: 832px;
  width: 90vw;
  z-index: 2;
}

.callout {
  padding: 20px;
  margin: 20px 0;
  border: 1px solid #eee;
  border-left-width: 5px;
  border-radius: 3px;

  h4 {
    margin-top: 0;
    margin-bottom: 5px;
  }

  p:last-child {
    margin-bottom: 0;
  }

  code {
    border-radius: 3px;
  }

  & + .bs-callout {
    margin-top: -5px;
  }
}

@each $name,$color in (default,#777),(primary,#428bca),
  (success,#5cb85c),(danger,#d9534f),(warning,#f0ad4e),
  (info,#5bc0de), (bdc,#29527a) {
  .callout-#{$name} {
    border-left-color: $color;

    h4 {
      color: $color;
    }
  }
}

.w-20 {
  width: 20% !important;
}

.table-borderless td,
.table-borderless th {
  border: 0;
  padding: 0.1em 0.5rem;

  .table td,
  .table th,
  .table a {
    font-size: 0.9rem;
  }
}

.cc-color-override-1434350510.cc-window {
  background-color: var(--brand4) !important;
}

iframe {
  width: 100%;
  height: 25em;
}

.font-nunito {
  font-family: Nunito, sans-serif;
}

.font-poppins {
  font-family: Poppins, sans-serif;
}

/* 
---------------------------------------------
Banner Style
--------------------------------------------- 
*/

.main-banner {
  position: relative;
  padding: 100px 0px 100px 0px;
}

.main-banner:after {
  content: url('/images/servers-min.png');
  top: 20px;
  right: 0;
  position: absolute;
  width: 950px;
  height: 746px;
  z-index: -1;
  overflow: hidden;
}

.main-banner::before {
  content: url('/images/banner-left.png');
  top: -1px;
  left: 0;
  position: absolute;
  width: 194px;
  height: 655px;
  z-index: -1;
}

.main-banner h6 {
  font-size: 20px;
  color: #5b03e4;
  text-transform: uppercase;
  font-weight: 700;
}

.main-banner .line-dec {
  margin: 30px 0px 20px 0px;
  width: 210px;
  height: 2px;
  background-color: #decdfa;
}

.main-banner h4 {
  font-size: 52px;
  color: #2a2a2a;
  font-weight: 700;
  line-height: 70px;
  margin-bottom: 20px;
}

.main-banner h4 em {
  font-style: normal;
  color: #A80D55;
}

.main-banner h4 span {
  color: #FF1464;
}

.main-banner p {
  padding-right: 25%;
  margin-bottom: 30px;
}

.main-banner .main-button {
  display: inline-block;
}

.main-banner span {
  display: inline-block;
  margin: 0px 10px 0px 10px;
  color: #7a7a7a;
}

.main-banner .second-button {
  display: inline-block;
}

.main-button a {
  font-size: 14px;
  color: #fff;
  background-color: #5b03e4;
  padding: 12px 30px;
  display: inline-block;
  border-radius: 25px;
  font-weight: 400;
  text-transform: capitalize;
  letter-spacing: 0.5px;
  transition: all .3s;
  position: relative;
  overflow: hidden;
}

.main-button a:hover {
  background-color: #5b03e4;
  color: #fff;
  opacity: 0.8;
}

.second-button a {
  font-size: 14px;
  color: #fff;
  background-color: #c03afe;
  padding: 12px 30px;
  display: inline-block;
  border-radius: 25px;
  font-weight: 400;
  text-transform: capitalize;
  letter-spacing: 0.5px;
  transition: all .3s;
  position: relative;
  overflow: hidden;
}

.second-button a:hover {
  background-color: #c03afe;
  color: #fff;
  opacity: 0.8;
}

  /* 
---------------------------------------------
Services Style
--------------------------------------------- 
*/

.services {
  position: relative;
  overflow: hidden;
}

.services .section-heading h2 {
  padding-right: 30px;
}

.services::before {
  content: url('/images/services.jpeg');
  top: 20px;
  left: 0;
  position: absolute;
  width: 844px;
  height: 714px;
  z-index: -1;
}

.services::after {
  content: url('/images/contact-left.jpg');
  bottom: -150px;
  right: 0;
  -webkit-transform: scaleX(-1);
  transform: scaleX(-1);
  position: absolute;
  width: 103px;
  height: 464px;
  z-index: -1;
}

.services .service-item {
  border: 1px solid #f3d7ff;
  border-radius: 23px;
  padding: 30px;
  background-color: #fff;
  margin-bottom: 30px;
  transition: all .4s;
}

.services .service-item h4 {
  font-size: 20px;
  color: #2a2a2a;
  margin-top: 20px;
  line-height: 30px;
  transition: all .4s;
}

.services .service-item:hover {
  border-color: #fff;
  box-shadow: 0px 0px 10px rgba(0, 0, 0, 0.15);
}

.services .service-item:hover h4 {
  color: #5b03e4;
}

@media (max-width: 1400px) {
  .main-banner .caption {
    background-color: rgba(250, 250, 250, 0.95);
    padding: 60px;
    border-radius: 23px;
  }

  .main-banner {
    padding: 20px 0px 60px 0px;
  }

  .main-banner p {
    padding-right: 0;
    margin-bottom: 0;
  }
}