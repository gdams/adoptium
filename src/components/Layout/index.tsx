import * as React from 'react'
import PropTypes from 'prop-types'
import { useStaticQuery, graphql } from 'gatsby'
import { SSRProvider } from '@react-aria/ssr'

import Navbar from '../Navbar'
import Banner from '../Banner'
import Footer from '../Footer'
import './layout.scss'

const Layout = ({ children }) => {
  const data = useStaticQuery(graphql`
    query SiteTitleQuery {
      site {
        siteMetadata {
          title
        }
      }
    }
  `)

  return (
    <SSRProvider>
      <div
        className='container-fluid p-0'
        style={{
          backgroundColor: 'var(--bg)',
          color: 'var(--text-normal)',
          transition: 'color 0.2s ease-out, background 0.2s ease-out'
        }}
      >
        <Navbar siteTitle={data.site.siteMetadata?.title || 'Title'} />
        <Banner />
        <main>
          <main>{children}</main>
        </main>
        <Footer />
      </div>
    </SSRProvider>
  )
}

Layout.propTypes = {
  children: PropTypes.node.isRequired
}

export default Layout
