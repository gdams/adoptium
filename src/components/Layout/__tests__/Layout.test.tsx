import React from 'react';
import { render } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest'
import Layout from '..';

describe('Layout component', () => {
  it('renders correctly with data', () => {
    const { container } = render(
      <Layout>
        mock-children
      </Layout>
    );
    expect(container).toMatchSnapshot();
  });
});
