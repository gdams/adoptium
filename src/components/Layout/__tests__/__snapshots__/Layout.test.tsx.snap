// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Layout component > renders correctly with data 1`] = `
<div>
  <div
    class="container-fluid p-0"
    style="transition: color 0.2s ease-out, background 0.2s ease-out;"
  >
    <div
      class="slice--navbar"
    />
    <div
      class="slice--banner"
    />
    <main>
      mock-children
    </main>
    <div
      class="slice--footer"
    />
  </div>
</div>
`;
